From bbfa9271e13752f72da3a67f1bc041164c5c8cdf Mon Sep 17 00:00:00 2001
From: Kevin McCarthy <kevin@8t8.us>
Date: Sat, 7 Dec 2019 13:47:37 -0800
Subject: Make shortpath/indent code guards explicit

In actuality, the parent_depth and indent_width won't ever be
negative.

But the new code for unset $sidebar_relative_shortpath_indent makes
this much less obvious.  Better to be explicit than have some small
future code change result in a bug.

Upstream-commit: https://gitlab.com/muttmua/mutt/commit/bbfa9271e13752f72da3a67f1bc041164c5c8cdf
Co-authored-by: Richard Russon <rich@flatcap.org>
---
 sidebar.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sidebar.c b/sidebar.c
index e4c258fc..df8ac38c 100644
--- a/sidebar.c
+++ b/sidebar.c
@@ -750,7 +750,7 @@ static void draw_sidebar (int num_rows, int num_cols, int div_width)
         }
       }
 
-      if (option (OPTSIDEBARSHORTPATH) && parent_depth)
+      if (option (OPTSIDEBARSHORTPATH) && (parent_depth > 0))
       {
         for (i = 0; parent_depth && sidebar_folder_name[i]; i++)
           if (strchr (SidebarDelimChars, sidebar_folder_name[i]))
@@ -758,7 +758,7 @@ static void draw_sidebar (int num_rows, int num_cols, int div_width)
         sidebar_folder_name += i;
       }
 
-      if (option (OPTSIDEBARFOLDERINDENT) && indent_width)
+      if (option (OPTSIDEBARFOLDERINDENT) && (indent_width > 0))
       {
         mutt_buffer_clear (indent_folder_name);
         for (i = 0; i < indent_width; i++)
-- 
2.24.1

